src/main/java/dev/thource/runelite/dudewheresmystuff/StorageManagerManager.java

```diff
@@ -159,6 +173,49 @@ public List<ItemStack> getItems() {
     return getStorages().filter(Storage::isEnabled).map(Storage::getItems).flatMap(List::stream).collect(Collectors.toList());
   }
 
+  /**
+   * Gets all known withdrawable items
+   *
+   * If the same item is in multiple storages, the item stacks are combined.
+   * "Same item" refers to items with the same canonical ID, but note that the
+   * actual ID of the stack will be set to the ID of one of the items
+   * arbitrarily. It is therefore recommended that callers do not use the IDs,
+   * only the canonical IDs.
+   *
+   * @return The item stacks
+   */
+  public Collection<ItemStack> getWithdrawableItems() {
+    // We need to deduplicate and combine item stacks if they're in multiple
+    // storages. This is a map from the stack's canonical (unnoted,
+    // un-placeholdered) ID to its stack.
+    TreeMap<Integer, ItemStack> items = new TreeMap<>();
+
+    getStorages()
+        .filter(Storage::isWithdrawable)
+        .map(Storage::getItems)
+        .flatMap(List::stream)
+        .forEach((ItemStack stack) -> {
+      if (stack.getQuantity() == 0 || stack.getId() == -1) {
+        return;
+      }
+
+      int id = stack.getCanonicalId();
+
+      ItemStack existing = items.get(id);
+      if (existing == null) {
+        items.put(id, stack);
```

Comment:
I'd make a copy of the itemstack here instead, then you don't need to worry about it in the else

Reply:
Was worried about performance, but that is simpler.